src/main/java/com/stuypulse/robot/MotorStall.java

```diff
@@ -0,0 +1,52 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package com.stuypulse.robot;
+
+import com.stuypulse.robot.subsystems.Funnel;
+
+import edu.wpi.first.wpilibj.Timer;
+
+public class MotorStall implements Runnable {
+        
+    private final Funnel m_Funnel;
+
+    private double start_encoder_value;
+    private double current_encoder_value;
+    private double change_distance;
+    private double encoder_approach_stall_threshold;
+    private int counter;
+
+    public MotorStall(Funnel funnel) {
+        encoder_approach_stall_threshold = 3.0;
+        m_Funnel = funnel;
+    }
+
+    @Override
+    public void run() {
+        checkStall();
+    }
+
+    public void checkStall() {
```

Comment:
Checkstall should go through an array of Subsystems that implement the interface mentioned above, and check to see if each is stalling